### PR TITLE
Protocol V2 docs

### PR DESCRIPTION
**Pull request by anomit**
Add documentation on the new components and approaches in the second version of the protocol

### Checklist
- [x] My branch is up-to-date with upstream/main branch.
- [x] Everything works and tested for major version of Python/NodeJS/Go and above.
- [x] I ran pre-commit checks against my changes.
- [x] I've written tests against my changes and all the current present tests are passing.

### Current behaviour

NA
### New expected behaviour

NA
#### Added 

Protocol V2 docs:
* Overview
* Sequencer
* Relayer
* Validator
* Protocol State

Fixes PowerLoom/product#141

## Deployment Instructions

NA

### CODE REVIEW

**Review by Seth-Schmidt — COMMENTED** [review]
Overall content is clear as far as highlighting the differences between v1 and v2 of the protocol. Starting with why the changes need to be made and breaking down each additional component into it's own section made it easy to follow as someone who hasn't use v2 yet.

I did find some minor changes to be made while going through it, made a secondary PR here:

https://github.com/PowerLoom/docs/pull/46

File: docs/Protocol/Protocol_v2/relay.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+which will keep a long-lived connection open. 
+
+Other peers will be able to dial through the relay peer using a 
+`p2p-circuit` address, which will forward traffic to its destination.
+```
+
+:::info
+[libp2p docs: Circuit Relay](https://docs.libp2p.io/concepts/nat/circuit-relay/)
+:::
+
```
Comment (Sulejman) [inline docs/Protocol/Protocol_v2/relay.md:35]: @anomit should we include trusted relayers here in this section now or later - this feature is currently in internal testing.

**Review by xadahiya — APPROVED** [review]
Looks good after https://github.com/PowerLoom/docs/pull/45/commits/735f4c96eed74ff446d909b713030586d6a467f1

**Review by Sulejman — APPROVED** [review]
LGTM, in sync with changes.

**Pull request merged**